File: Q1-100/Q1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> table = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            int another = target - num;
            if (table.containsKey(another)) {
                return new int[] { table.get(another), i };
            } else {
                table.put(num, i);
            }
        }
        return new int[0];
    }
}

File: Q1-100/Q1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> table;
        for (int i = 0; i < nums.size(); i++) {
            int num = nums[i];
            int another = target - num;
            if (table.count(another)) {
                return {table[another], i};
            } else {
                table.insert({num, i});
            }
        }
        return {};
    }
}